src/module.c

```diff
@@ -3319,19 +3339,38 @@ RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   RM_TRY(RMCreateSearchCommand(ctx, "FT.INFO", SafeCmd(InfoCommandHandler), "readonly", 0, 0, -1, ""))
   RM_TRY(RMCreateSearchCommand(ctx, "FT.SEARCH", SafeCmd(DistSearchCommand), "readonly", 0, 0, -1, "read"))
   RM_TRY(RMCreateSearchCommand(ctx, "FT.PROFILE", SafeCmd(ProfileCommandHandler), "readonly", 0, 0, -1, "read"))
+
+  int cursorFirst, cursorLast, cursorStep;
   if (clusterConfig.type == ClusterType_RedisLabs) {
-    RM_TRY(RMCreateSearchCommand(ctx, "FT.CURSOR", SafeCmd(CursorCommand), "readonly", 3, 1, -3, "read"))
+    cursorFirst = 3;
+    cursorLast = 1;
+    cursorStep = -3;
   } else {
-    RM_TRY(RMCreateSearchCommand(ctx, "FT.CURSOR", SafeCmd(CursorCommand), "readonly", 0, 0, -1, "read"))
-  }
+    cursorFirst = 0;
+    cursorLast = 0;
+    cursorStep = -1;
+  }
+  RM_TRY(RMCreateSearchCommand(ctx, "FT.CURSOR", NULL, "readonly", cursorFirst, cursorLast, cursorStep, "read"));
+  RedisModuleCommand *cursorCmd = RedisModule_GetCommand(ctx, "FT.CURSOR");
+  // Register "READ" and "DEL" subcommands to run from the coordinator thread pool
+  RM_TRY(RedisModule_CreateSubcommand(cursorCmd, "READ", SafeCmd(CursorReadCommand), "readonly", cursorFirst, cursorLast, cursorStep));
+  RM_TRY(RedisModule_CreateSubcommand(cursorCmd, "DEL", SafeCmd(CursorDelCommand), "readonly", cursorFirst, cursorLast, cursorStep));
+  // Register "GC" subcommand as the local handler (exactly like its `_FT` variants)
+  RM_TRY(RedisModule_CreateSubcommand(cursorCmd, "GC", SafeCmd(RSCursorGCCommand), "readonly", cursorFirst, cursorLast, cursorStep));
```

Comment:
That's a good point. We may want to consider always using a BG thread for the GC command (maybe with the clean pool?)

Reply:
I think we should, this will also save the need of finding a solution for potential problems like in MOD-8108. 
`cleanPool` is an interesting idea, sounds good to me.